cpp/core/jni/JniWrapper.cc

```diff
@@ -148,6 +149,69 @@ class JavaInputStreamAdaptor final : public arrow::io::InputStream {
   bool closed_ = false;
 };
 
+class JavaInputStreamVeloxWrapper final : public JavaInputStreamWrapper {
```

Comment:
Can we use `JavaInputStreamAdapter`?

Reply:
The existing `JavaInputStreamAdapter` is defined here https://github.com/apache/incubator-gluten/blob/182a0299993b38e04ee053158f71b2500c59eb09/cpp/core/jni/JniWrapper.cc#L79 Can we use this class?

Reply:
@marin-ma This class can't be used because JavaInputStreamVeloxWrapper wraps Velox's stream and can't be received by arrow's ```arrow::io::InputStream```.

Reply:
@marin-ma done